src/v/pandaproxy/schema_registry/avro.cc

```diff
@@ -132,19 +134,23 @@ result<void> sanitize(
   rapidjson::GenericValue<rapidjson::UTF8<>>::Array& a,
   rapidjson::MemoryPoolAllocator<>& alloc);
 
-result<void> sanitize_name(
+result<std::string_view> sanitize_union_symbol_name(
```

Comment:
This change was not intentional.

Reply:
is that something you are going to change? while you are in there, it might be worth leaving a comment that npos+1 is zero

Reply:
Return type fixed and `npos + 1 == 0` is no longer assumed.